src/middlewared/middlewared/plugins/pool.py

```diff
@@ -3120,6 +3148,7 @@ def add(attr):
         ('edit', _add_inherit('readonly')),
         ('edit', _add_inherit('recordsize')),
         ('edit', _add_inherit('snapdir')),
+        ('edit', _user_props_rename('user_properties')),
```

Comment:
This will not allow to replace dataset's properties with ones specified in the request, won't it?

Reply:
It would ?

Reply:
How? Your `edit` seems to be renaming `user_properties` into `user_properties_update`